style-guide/styles/wds-components/_buttons.scss

```diff
@@ -98,6 +100,28 @@
 		}
 	}
 
+	&.wds-has-icon {
+		padding-left: $button-padding-horizontal + $wds-icon-small-size + 6px;
+		position: relative;
+
+		&.wds-is-squished {
+			padding-left: $button-squished-padding-horizontal + $wds-icon-tiny-size + 6px;
```

Comment:
Can we make this solution more bullet proof by forcing width and height of the icon?